src/transformers/models/nucleus_x/configuration_nucleus_x.py

```diff
@@ -0,0 +1,211 @@
+            The non-linear activation function (function or string) in the decoder.
+        dropout (`float`, *optional*, defaults to 0.0):
+            Dropout probability
+        activation_dropout (`float`, *optional*, defaults to 0.0):
+            Dropout probability after activation in FFN.
+        drop_path_rate (`float`, *optional*, defaults to 0.0):
+            Drop Path (or Stochastic Depth, https://arxiv.org/abs/1603.09382) rate. Uses timm implementation (see
+            https://github.com/huggingface/pytorch-image-models).
+        decoder_embed_dim (`int`, *optional*, defaults to 4096):
+            Decoder embedding dimension.
+        decoder_value_embed_dim (`int`, *optional*, defaults to 6912):
+            Decoder value embedding dimension.
+        decoder_ffn_embed_dim (`int`, *optional*, defaults to 6912):
+            Decoder embedding dimension for FFN.
+        decoder_layers (`int`, *optional*, defaults to 32):
+            Number of hidden layers in the Transformer encoder.
+        decoder_retention_heads (`int`, *optional*, defaults to 16):
+            Number of decoder retention heads.
+        decoder_normalize_before (`bool`, *optional*, defaults to `True`):
+            Apply rms_norm before each decoder block
+        rms_norm_embedding (`bool`, *optional*, defaults to `True`):
+            Add rms_norm to embedding
+        no_scale_embedding (`bool`, *optional*, defaults to `False`):
+            If True, dont scale embeddings
+        recurrent_chunk_size (`int`, *optional*, defaults to 512):
+            The chunk size for `"chunkwise"` mode.
+        use_lm_decay (`bool`, *optional*, defaults to `False`):
+            Whether to use language model decay. (Found in https://arxiv.org/pdf/2307.08621.pdf, Page 7, Section 3.1,
+            Paragraph **Parameter Allocation**, last sentence)
+        z_loss_coeff (`float`, *optional*, defaults to 0.0):
+            coefficient for z-loss (Used in PaLM, https://arxiv.org/pdf/2204.02311.pdf)
+        deepnorm (`bool`, *optional*, defaults to `False`):
+            DeepNorm (https://arxiv.org/abs/2203.00555). Disables `decoder_normalize_before` and `subln`.
+        subln (`bool`, *optional*, defaults to `True`):
+            SubLN (https://arxiv.org/abs/2210.06423). Enables `decoder_normalize_before` and disables `deepnorm`.
+        rms_norm_eps (`float`, *optional*, defaults to 1e-06):
+            The epsilon used by the rms normalization layers.
+        groupnorm_eps (`float`, *optional*, defaults to 1e-06):
+            The epsilon used by the group_norm layer in NucleusXMultiScaleRetention.
+        tie_word_embeddings (`bool`, *optional*, defaults to `False`):
+            Whether to tie weight embeddings
+
+
+    ```python
+    >>> from transformers import NucleusXModel, NucleusXConfig
+
+    >>> # Initializing a NucleusX-7B style configuration
+    >>> configuration = NucleusXConfig()
```

Comment:
```suggestion
    >>> configuration = NucleusXConfig(decoder_layers=2)
```
Also, one more comment! The doctest runner is crashing on this file, and I suspect the reason is that it's running out of memory because you're initializing a 7B model in float32 and so using 28GB of memory, which is a lot for the doctest runner! Maybe change this line to initialize a much smaller model?

Reply:
I was trying to find out why the doctest is failing for a long time, and this makes total sense!